### ISSUE TITLE
BLE connection gives an exception: Cannot read property 'localName' of undefined

### ISSUE THREAD

**Issue by koktaildotcom** [athombv/apps-sdk-issues#7]
```
TypeError: Cannot read property 'localName' of undefined
    at new BleAdvertisement (/opt/homey-client/system/manager/ManagerApps/bootstrap/sdk/v2/lib/BleAdvertisement.js:1:369)
    at ManagerBLE.__createAdvertisementInstance (/opt/homey-client/system/manager/ManagerApps/bootstrap/sdk/v2/manager/ble.js:1:1220)
    at __advertisementsByPeripheralUUID.(anonymous function).__client.emit.then.e (/opt/homey-client/system/manager/ManagerApps/bootstrap/sdk/v2/manager/ble.js:1:1089)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
```

**Comment by WeeJeWel** [comment]
Fixed in Homey v2.0.0-rc.3

**Comment by koktaildotcom** [comment]
This is not fixed in v2.0.0-rc.3

**Comment by WeeJeWel** [comment]
Can you share more code when this happens?

**Comment by koktaildotcom** [comment]
When i'm trying to find an BleAdvertisement from the ManagerBLE.

where let uuid = `<string>`
```
Homey.ManagerBLE.find(uuid).then(function (advertisement) {
    return advertisement;
});
```

**Comment by koktaildotcom** [comment]
Sorry. The update to rc3 was failed and i've retested it on rc2.

**Comment by WeeJeWel** [comment]
No problem!

**Comment by koktaildotcom** [comment]
This issue is back at v2.0.0-rc.8

**Comment by koktaildotcom** [comment]
I've added the code where this happens to: https://github.com/koktaildotcom/homey-mi-flora/tree/feature/ble-advertisement-exception.

note: Due to compatibility to 2.0 i've added BLE permissions. They should be removed to be compatible with < 2.0

**Comment by koktaildotcom** [comment]
@WeeJeWel any idea when you can take a look at this issue?

**Comment by koktaildotcom** [comment]
Still an issue with 2.0.0

**Comment by WeeJeWel** [comment]
This has been fixed in Homey v2.0.5. Thank you for reporting!

**Comment by koktaildotcom** [comment]
This is still an issue on Homey 2.0.5

**Comment by WeeJeWel** [comment]
Can you provide a reproduction path?

**Comment by koktaildotcom** [comment]
Looks related to: https://github.com/athombv/homey-apps-sdk-issues/issues/73

**Comment by koktaildotcom** [comment]
> Can you provide a reproduction path?

1. Checkout: https://github.com/koktaildotcom/homey-mi-flora/tree/feature/ble-advertisement-exception. and run the code.
1. Set the update interval to 1000 here: https://github.com/koktaildotcom/homey-mi-flora/blob/ddac6bebc3417fc00733065b2df2d4a012eadb49/lib/MiFloraDriver.js#L61 (otherwise you need to wait it out)
1. Add a new device
1. Wait until the second iteration kicks in, this is the point the error occur.

**Comment by WeeJeWel** [comment]
Thank you! That helped a lot. I'm 100% sure I found the issue and fixed it in Homey v2.0.5-rc.5.
